test/groups/test_special_unitary_group.jl

```diff
@@ -0,0 +1,41 @@
+using LieGroups, Random, Test
+
+s = joinpath(@__DIR__, "..", "LieGroupsTestSuite.jl")
+!(s in LOAD_PATH) && (push!(LOAD_PATH, s))
+using LieGroupsTestSuite
+
+@testset "Special Unitary Group" begin
+    G = SpecialUnitaryGroup(2)
+    #g1 = 1 / sqrt(2) * [1.0 1.0; -1.0 1.0]
+    #g2 = [0.0 -1.0; 1.0 0.0]
+    #g3 = [1.0 0.0; 0.0 1.0]
+    #X1, X2, X3 = [1.0 0.0; 0.0 0.0], [0.0 0.0; 0.0 1.0], [0.0 0.5; 0.5 0.0]
+    properties = Dict(
```

Comment:
We should have some tests for different operations here I think, not just `show`.

Reply:
Sure, as I wrote above, test coverage might not yet be fully complete, but I work on it literally every day to improve that a bit. Feel free to add tests as well.
Over the weekend I managed to cover some 170+ lines of code, but sure there is both about 100+ lines and a few semantic tests left.

Reply:
Hm, this was a good note, but for the last 30 minutes I just see errors increasing again. So both this and the unitary group probably still need quiiiiiiiiiiite some work, probably especially allocation magic is off often and allocates real stuff when it should allocate complex.

Reply:
Ok, the main things not working are `hat`/ `vee` because we do not have any basis implemented in the complex case for general unitary (especially no ONB that the LieAlgebraBasis wants to fall back to) – and `rand` also due to missing support on Manifolds. Is that ok to first fix those over there and come back here on a later PR?

Reply:
I think we can skip `hat` and `vee` for now.